dev/com.ibm.ws.microprofile.faulttolerance.2.0/test/src/com/ibm/ws/microprofile/faulttolerance20/state/impl/CircuitBreakerStateImplTest.java

```diff
@@ -108,6 +108,91 @@ public void testFailOn() throws InterruptedException {
         runSuccesses(state, 1);
     }
 
+    @Test
+    public void testSkipOn() throws InterruptedException {
+        CircuitBreakerPolicyImpl policy = new CircuitBreakerPolicyImpl();
+        policy.setDelay(Duration.ofMillis(100));
+        policy.setFailureRatio(0.75);
+        policy.setRequestVolumeThreshold(4);
+        policy.setSuccessThreshold(1);
+        policy.setFailOn(TestExceptionA.class);
+        policy.setSkipOn(TestExceptionAextended.class);
+
+        // Summary:
+        // TestExceptionA         counts as a failure
+        // TestExceptionAextended counts as a pass due to skipOn despite being instance of failOn
+        // TestExceptionB         counts as a pass since not instance of failOn
+        // circuit opens if 75% of last 4 requests failed
+        // circuit half-closes after 100ms
+        // circuit fully closes after 1 successful request
+
+        // 4x TestExceptionA -> 100% failure rate -> circuit open
+        CircuitBreakerStateImpl state = new CircuitBreakerStateImpl(policy, dummyMetrics);
+        runFailures(state, 4, TestExceptionA.class);
+        assertFalse(state.requestPermissionToExecute());
+
+        // Wait >150ms -> circuit half-open
+        Thread.sleep(150);
+        // Run 1 success, results in circuit closed
+        runSuccesses(state, 1);
+
+        // 4x TestExceptionAextended -> 0% failure rate -> circuit closed
+        runFailures(state, 4, TestExceptionAextended.class);
+        // assert circuit closed
+        runSuccesses(state, 1);
+
+        // 4x TestExceptionB -> 0% failure rate -> circuit closed
+        // Since TestExceptionB (or any of its superclasses) isn't mentioned in either failOn nor skipOn
+        runFailures(state, 4, TestExceptionB.class);
+        // assert circuit closed
+        runSuccesses(state, 1);
+    }
+
+    @Test
+    public void testMultipleExceptions() throws InterruptedException {
```

Comment:
It doesn't need any changes, but it's worth noting that this test includes an important case which wasn't in `testSkipOn`.

The important case is `failOn=TestExceptionBextended, skipOn=TestExceptionB` which asserts that `skipOn` always takes priority over `failOn`, even if an exception in `skipOn` is a superclass of an exception in `failOn`.